server/rest/src/main/java/org/infinispan/rest/resources/ProtobufResource.java

```
@@ -0,0 +1,233 @@
+
+
+      NettyRestResponse.Builder builder = new NettyRestResponse.Builder();
+
+      CompletableFuture<Object> putSchema;
+      if (create) {
+         putSchema = cache.putIfAbsentAsync(schemaName, contents.asString()).thenApply(result -> {
+            if (result == null) {
+               builder.status(HttpResponseStatus.CREATED);
+            } else {
+               builder.status(HttpResponseStatus.CONFLICT);
+            }
+            return result;
+         });
+      } else {
+         putSchema = cache.replaceAsync(schemaName, contents.asString()).thenApply(result -> {
+            if (result == null) {
+               builder.status(HttpResponseStatus.NOT_FOUND);
+            } else {
+               builder.status(HttpResponseStatus.OK);
+            }
+            return result;
+         });
+      }
+
+      return putSchema
+            .thenCompose(r -> {
+               if(isOkOrCreated(builder)) {
+                  return cache.getAsync(schemaName + ProtobufMetadataManagerConstants.ERRORS_KEY_SUFFIX);
+               } else {
+                  return CompletableFuture.completedFuture(null);
+               }
+            })
+            .thenApply(validationError -> {
+               if(isOkOrCreated(builder)) {
+                  ProtoSchema protoSchema = new ProtoSchema();
+                  protoSchema.name = schemaName;
+                  if(validationError != null) {
+                     protoSchema.error = createErrorContent(schemaName, (String) validationError);
+                     builder.status(HttpResponseStatus.BAD_REQUEST);
+                  }
+                  byte[] detailsResponse = new byte[0];
+                  try {
+                     detailsResponse = invocationHelper.getMapper().writeValueAsBytes(protoSchema);
+                  } catch (JsonProcessingException e) {
+                     builder.status(HttpResponseStatus.INTERNAL_SERVER_ERROR);
```

Do we want to at least log the exception?

---

I change this launching an exception, but should not be happening 
it's uneven across resources now, sometimes we launch a throw new RestResponseException(e); and others just catch

---

I understand that different failures may be propagated. But I think we should at least log an exception instead of just catching it and making it impossible to debug.

---

And sorry the previous code doing this should at least log the error as a trace as well. We should not be swallowing exceptions, except for in very rare cases (especially when we can be reporting an error to the user).

---

there is a global issue concerning logging in the REST resources.
Can we discuss this too with @gustavonalle so we can do some follow up PR on error handling and logging strategy, please?

---

There is an error JSON document to report errors, the query endpoint is using it

---

Some endpoints are not implementing this to handle ```JsonProcessingException``` as it is very unlikely this error would be thrown since the JSON is mapped via Jackson based on always the same class

---

I don't think just adding a single trace message needs another PR... And I have heard many a developer say something is very unlikely to impossible, including myself, and it ends up  happening. So I would rather have at least a trace message when I am completely ignoring a checked Exception.

---

Ok, I agree adding log won't hurt  :smile: